hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/JsonUpsertTarget.java

```diff
@@ -16,131 +16,26 @@
-                        case BIGINT:
-                            generator.writeNumberField(path, (Long) value);
-                            break;
-                        case REAL:
-                            generator.writeNumberField(path, (Float) value);
-                            break;
-                        case DOUBLE:
-                            generator.writeNumberField(path, (Double) value);
-                            break;
-                        case DECIMAL:
-                        case TIME:
-                        case DATE:
-                        case TIMESTAMP:
-                        case TIMESTAMP_WITH_TIME_ZONE:
-                        case VARCHAR:
-                            generator.writeStringField(path, (String) QueryDataType.VARCHAR.convert(value));
-                            break;
-                        case OBJECT:
-                            injectObject(path, value);
-                            break;
-                        default:
-                            throw QueryException.error("Unsupported type: " + type);
-                    }
-                }
-            } catch (IOException e) {
-                throw sneakyThrow(e);
-            }
-        };
-    }
-
-    private void injectObject(String path, Object value) throws IOException {
-        generator.writeFieldName(path);
-        if (value == null) {
-            generator.writeNull();
-        } else if (value instanceof TreeNode) {
-            generator.writeTree((TreeNode) value);
-        } else if (value instanceof Map) {
-            generator.writeObject(value);
-        } else if (value instanceof Boolean) {
-            generator.writeBoolean((boolean) value);
-        } else if (value instanceof Byte) {
-            generator.writeNumber((byte) value);
-        } else if (value instanceof Short) {
-            generator.writeNumber((short) value);
-        } else if (value instanceof Integer) {
-            generator.writeNumber((int) value);
-        } else if (value instanceof Long) {
-            generator.writeNumber((long) value);
-        } else if (value instanceof Float) {
-            generator.writeNumber((float) value);
-        } else if (value instanceof Double) {
-            generator.writeNumber((double) value);
-        } else {
-            generator.writeString((String) QueryDataType.VARCHAR.convert(value));
-        }
-    }
-
-    @Override
-    public void init() {
-        baos.reset();
-        try {
-            generator = JSON_FACTORY.createGenerator(baos);
-            generator.writeStartObject();
-        } catch (IOException e) {
-            throw sneakyThrow(e);
-        }
-    }
-
     @Override
     public Object conclude() {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
         try {
-            generator.writeEndObject();
-            generator.close();
+            ObjectOutputStream oos = new ObjectOutputStream(baos);
```

Comment:
This doesn't write the bytes for the JSON document, but some garbage. Tests fail then.